README.md

```diff
@@ -1 +1,1148 @@
+			              ||    .'
+			               \\  (
+			                >\  >
+			            ,.-' >.'
+			           <.'_.''
+			             <'
+```
+
+This will package up and then push the code from the local directory to the
+balena builders, which examines any `Dockerfile`, `Dockerfile.template`,
+`docker-compose.yml` or `package.json` file to determine how to build the code
+and then create a Docker image that will be stored in the balenaCloud private
+Docker repository.
+
+The `push` command actually accepts a wide range of arguments, but this is the
+basic (and most widely used) form of the command.
+
+As a balenaFin has been provisioned against the `cliApp` application, you'll
+soon see that it starts to download the code we've just pushed (by observing
+the dashboard). Eventually, the application image will be pulled and the
+Supervisor on the device will start the service. You can look at the logs by
+using the following balenaCLI command:
+```
+$ balena logs 1234567
+[Logs]    [9/3/2019, 11:18:50 AM] Downloading image 'registry2.balena-cloud.com/v2/1234568b5d909bf95c2bbe6c1df150f3@sha256:75f4f5bfb6e610f2b7fe8d61e8a77979a4be3de344cad09fba4df5bba8da73a3'
+[Logs]    [9/3/2019, 11:18:51 AM] Service exited 'main sha256:f9813f5cc0ee22ca166c93c2c72e2fb6f37dff6cb4add332e3f57e418de6b339'
+[Logs]    [9/3/2019, 11:18:58 AM] Downloaded image 'registry2.balena-cloud.com/v2/1234568b5d909bf95c2bbe6c1df150f3@sha256:75f4f5bfb6e610f2b7fe8d61e8a77979a4be3de344cad09fba4df5bba8da73a3'
+[Logs]    [9/3/2019, 11:18:58 AM] Killing service 'main sha256:f9813f5cc0ee22ca166c93c2c72e2fb6f37dff6cb4add332e3f57e418de6b339'
+[Logs]    [9/3/2019, 11:18:58 AM] Killed service 'main sha256:f9813f5cc0ee22ca166c93c2c72e2fb6f37dff6cb4add332e3f57e418de6b339'
+[Logs]    [9/3/2019, 11:18:58 AM] Deleting image 'registry2.balena-cloud.com/v2/12344fad6b184fb68e1f94f8fced204d@sha256:35428c9df601789509f2100a8e6a22c2a2f0257bb49308053cb598be82e67ce4'
+[Logs]    [9/3/2019, 11:18:58 AM] Deleted image 'registry2.balena-cloud.com/v2/12344fad6b184fb68e1f94f8fced204d@sha256:35428c9df601789509f2100a8e6a22c2a2f0257bb49308053cb598be82e67ce4'
+[Logs]    [9/3/2019, 11:18:59 AM] Installing service 'main sha256:e2683253db679de87adca3b3fa80ac6aa16b62be4329524c394d3ce0d5f96f65'
+[Logs]    [9/3/2019, 11:19:00 AM] Installed service 'main sha256:e2683253db679de87adca3b3fa80ac6aa16b62be4329524c394d3ce0d5f96f65'
+[Logs]    [9/3/2019, 11:19:00 AM] Starting service 'main sha256:e2683253db679de87adca3b3fa80ac6aa16b62be4329524c394d3ce0d5f96f65'
+[Logs]    [9/3/2019, 11:19:01 AM] Started service 'main sha256:e2683253db679de87adca3b3fa80ac6aa16b62be4329524c394d3ce0d5f96f65'
+[Logs]    [9/3/2019, 11:19:02 AM] [main]
+[Logs]    [9/3/2019, 11:19:02 AM] [main] > balena-cli-masterclass@1.0.0 start /usr/src/app
+[Logs]    [9/3/2019, 11:19:02 AM] [main] > node src/helloworld.js; sleep infinity
+[Logs]    [9/3/2019, 11:19:02 AM] [main]
+[Logs]    [9/3/2019, 11:19:03 AM] [main] Hello world!
+```
+
+### 3.2 `git push`
```

Comment:
To me, this feels a little outside the scope of a balenaCLI masterclass. There's definitely value in debugging CLI issues/providing alternatives & workarounds to end users, but if it's "legacy" as well perhaps we shouldn't advertise it.

Reply:
I was very on the fence about this, but we still have a lot of customers using it and it seemed the right thing to do for completeness. @shaunmulligan what are your thoughts on this?

Reply:
@hedss yeah, its a difficult one, I would almost want this in a CI/CD type master class or some kind of "alternative" deployment methods. I guess we will have more soon with the abilty to deploy from a github repo too.

Reply:
@shaunmulligan Yeah, I changed this around a bit, and I think once we have a CI/CD based class, this can get removed so `balena push` is the only 'official' way on the CLI.